test/test_dynamic_shapes.py

```diff
@@ -388,5 +395,145 @@ def forward(self, a_1: f32[s0, s1], b_1: f32[s2, s1]):
         return (getitem, getitem_1)""")  # noqa: B950
 
 
+# This environment variable controls whether or not we print expected failure
+# lists at the end of a test suite run.  The intended usage looks like this:
+#
+# 1. Run `PYTORCH_COLLECT_EXPECT=1 python test/test_dynamic_shapes.py -k TestSymNumberMagicMethods`.
+# 2. Given the printed xfail list, add them to the set expected_failure_sym_magic_methods.
+COLLECT_EXPECT = os.getenv('PYTORCH_COLLECT_EXPECT', '0') == '1'
+
+seen_failed = []
+def print_seen():
+    out = []
+    for key, reason in seen_failed:
+        # Make sure the generated line is lint clean
+        out.append(f"    {key},  # {reason}"[:120])
+
+    print("expected_failure_sym_magic_methods = {")
+    print("\n".join(out))
+    print("}")
+
+if COLLECT_EXPECT:
+    atexit.register(print_seen)
+
+expected_failure_sym_magic_methods = {
```

Comment:
All of these should be fairly easy to fix but this PR was getting too big already.